pkg/build/builder/cmd/scmauth/util.go

```diff
@@ -59,3 +60,41 @@ func ensureGitConfigIncludes(path string, context SCMAuthContext) error {
 	content := []byte(strings.Join(lines, "\n"))
 	return ioutil.WriteFile(gitconfig, content, 0600)
 }
+
+func EnsureGitConfigIncludesProxyConfig(context SCMAuthContext, gitSource *buildv1.GitBuildSource) error {
```

Comment:
Need a unit test verifying the behavior of this function.

Reply:
unit test pushed @adambkaplan